openhands/server/middleware.py

```
@@ -75,6 +75,8 @@ def _clean_old_requests(self, key: str) -> None:
         self.history[key] = [ts for ts in self.history[key] if ts > cutoff]
 
     async def __call__(self, request: Request) -> bool:
+        if not request.client:
+            return True  # Allow requests without client info
```

I'm not very sure if this is ideal behavior here, would it be better to raise an exception?

---

It should be okay, but it might not matter because we're thinking to remove [the rate limiting here](https://github.com/All-Hands-AI/OpenHands/pull/6360)